### PR TITLE
api: use heartbeat status to make store state more clear

### LINKED ISSUE #1566: Misleading store state
Currently, the status of down/disconnect is not reflected in the meta information of the store. As long as you do not manually go offline, the meta is always Up. This leads to the inability to determine the down state when using the grpc or HTTP interface to query the meta information of the store, and it is easy to mistake it as normal (but not).

Proposal: Define new state to use in the API, distinguish from that persistent to storage.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

close #1566

### What is changed and how it works?

This PR separates store state and heartbeat status and add two more heartbeat status.

### Check List

Tests

- Unit test

Code changes

- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn): 

### Release note

```release-note
Use an individual heartbeat status for the store.
```

### CODE REVIEW

File: server/api/store.go [hunk 1]
```diff
@@ -117,8 +118,10 @@ type StoreInfo struct {
 }
 
 const (
-	disconnectedName = "Disconnected"
-	downStateName    = "Down"
+	alive        = "Alive"
+	disconnected = "Disconnected"
+	down         = "Down"
+	dead         = "Dead"
```
Comment (disksing) [inline server/api/store.go:124]: I'm not sure if it is a good name. It may lead to some misunderstandings, as the tikv process may still be running.

Comment (rleungx) [reply server/api/store.go:124]: When it enters the dead status, the TiKV has already been Tombstone.

**Review by disksing — APPROVED** [review]

File: server/api/store.go (hunk 1 above)
Comment (disksing) [reply server/api/store.go:124]: May need some opinions from other reviewers.

Comment (bufferflies) [reply server/api/store.go:124]: I think the down and disconnect is fuzzy for user if they do not see the code, and they may not known down will step to dead after the store has no regions.

Comment (JmPotato) [reply server/api/store.go:124]: `Dead` is kind of arbitrary. Maybe a word like `Removed`, `Outcasted`, or `Detached` is a better choice to indicate that the TiKV node does not belong to the cluster anymore.

**Pull request closed without merging**